src/network/networkd-varlink.c

```diff
@@ -1,6 +1,158 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 
 #include "networkd-varlink.h"
+#include "alloc-util.h"
+#include "fd-util.h"
+#include "json.h"
+#include "varlink.h"
+
+typedef struct LookupParameters {
+        int ifindex;
+} LookupParameters;
+
+static int open_lldp_neighbors(int ifindex, FILE **ret) {
+        _cleanup_free_ char *p = NULL;
+        FILE *f;
+
+        if (asprintf(&p, "/run/systemd/netif/lldp/%i", ifindex) < 0)
+                return -ENOMEM;
+
+        f = fopen(p, "re");
```

Comment:
nah, you are inside of networkd's address space here. please read the data directly from the data structures already in memory.